website_menu_by_user_status/models/website_menu.py

```diff
@@ -14,27 +14,28 @@ class WebsiteMenu(models.Model):
     user_logged = fields.Boolean(
         string="User Logged",
         default=True,
-        help=_("If checked, "
-               "the menu will be displayed when the user is logged "
-               "and give access.")
+        help="If checked, "
+        "the menu will be displayed when the user is logged "
+        "and give access.",
     )
 
     user_not_logged = fields.Boolean(
         string="User Not Logged",
         default=True,
-        help=_("If checked, "
-               "the menu will be displayed when the user is not logged "
-               "and give access.")
+        help="If checked, "
+        "the menu will be displayed when the user is not logged "
+        "and give access.",
     )
 
-    @api.one
+    @api.multi
     def _compute_visible(self):
         """Display the menu item whether the user is logged or not."""
-        super()._compute_visible()
-        if not self.is_visible:
-            return
-
-        if self.env.user == self.env.ref('base.public_user'):
-            self.is_visible = self.user_not_logged
-        else:
-            self.is_visible = self.user_logged
+        for menu in self:
+            super(WebsiteMenu, menu)._compute_visible()
```

Comment:
Only call once to super, as it was done before.

Reply:
but odoo 11 is still using this in core:
```python
    @api.one
    def _compute_visible(self):
        visible = True
        if self.page_id and not self.page_id.sudo().is_visible and not self.user_has_groups('base.group_user'):
            visible = False
        self.is_visible = visible
```

wouldn't it cause problems with multi records?

Reply:
Not, because it's decorated with `@api.one`, which automatically does the loop.